internal/kafka/internal/handlers/validation.go

```diff
@@ -440,19 +440,6 @@ func validateEnterpriseClusterEligibleForDeregistration(ctx context.Context, clu
 		if cluster.ClusterType != api.EnterpriseDataPlaneClusterType.String() {
 			return errors.Forbidden("unable to deregister cluster whose type is not: %s", api.EnterpriseDataPlaneClusterType.String())
 		}
-		if !forceDeleteFlag {
-			clusterKafkaCount, err := clusterService.FindKafkaInstanceCount([]string{clusterID})
```

Comment:
should the check for non-empty cluster stay here and fail if there are kafkas on the cluster?

Reply:
+1